doc/callbacks.md

```diff
@@ -136,14 +136,73 @@ EarlyStopAtMinLoss cb(
       // You could also, e.g., print the validation loss here to watch it converge.
       return lrfValidation.Evaluate(coordinates);
     });
-    
+
 arma::mat coordinates = lrfTrain.GetInitialPoint();
 SMORMS3 smorms3;
 smorms3.Optimize(lrfTrain, coordinates, cb);
 ```
 
 </details>
 
+### GradClipByNorm
+
+Given the gradient, and a maximum norm value, the callback normalizes the
+gradient so that its L2-norm is less than or equal to the given maximum norm
+value.
```

Comment:
It might be useful to add a sentence or two about where and when this is useful.  (Same for `GradClipByValue`.) :+1: